cdap-docs/developers-manual/source/building-blocks/workflows.rst

```diff
@@ -274,14 +477,108 @@ otherwise, the other path will be taken. The diagram for this code would be:
+    //                                        True  |                          |   
+    // Start --> RecordVerifier --> Predicate ----->|                          |----> StatusReporter --> End    
+    //                                  |           |                          |  |
+    //                                  | False     |--> PurchaseByProduct --->|  |
+    //                                  |                                         |
+    //                                  |------------> ProblemLogger ------------>|
+ 
+    // Use case 1: Predicate can add the key "branch" in the WorkflowToken with value as
+    // "true" if true branch will be executed or "false" otherwise. In "StatusReporter" in
+    // order to get which branch in the Workflow was executed, use:
+    
+    boolean bTrueBranch = token.getAsBoolean("branch");
+ 
+    // Use case 2: You may want to compare the records emitted by "PurchaseByCustomer"
+    // and "PurchaseByProduct", in order to find which job is generating more records:
+    
+    String flattenReduceOutputRecordsCounterName = "org.apache.hadoop.mapreduce.TaskCounter.REDUCE_OUTPUT_RECORDS";
+    String purchaseByCustomerCounterValue = token.get(flattenReduceOutputRecordsCounterName, "PurchaseByCustomer", 
+                                                      WorkflowToken.Scope.SYSTEM);
+    String purchaseByProductCounterValue = token.get(flattenReduceOutputRecordsCounterName, "PurchaseByProduct", 
+                                                     WorkflowToken.Scope.SYSTEM);
+  
+    // Use case 3: Since Workflow can have multiple complex conditions and forks in its
+    // structure, in the "StatusReporter", you may want to know how many actions were
+    // executed as a part of a run. If the number of nodes executed were below a certain
+    // threshold, send an alert. Assuming that every node in the Workflow adds the key
+    // "action.type" with the value as action type for that node in the WorkflowToken,
+    // you can determine the breakdown by action type in a particular Workflow run:
+    
+    List<NodeValueEntry> nodeValues = token.getAll("action.type");
+    int totalNodeExecuted = nodeValues.size();
+    int mapReduceNodes = 0;
+    int sparkNodes = 0;
+    int customActionNodes = 0;
+    int conditions = 0;
+    for (NodeValueEntry entry : nodeValues) {
+      if (entry.getValue().equals("MAPREDUCE")) {
+        mapReduceNodes++;
+      }
+      if (entry.getValue().equals("SPARK")) {
+        sparkNodes++;
+      }
+      if (entry.getValue().equals("CUSTOM_ACTION")) {
+        customActionNodes++;
+      }
+      if (entry.getValue().equals("CONDITION")) {
+        conditions++;
+      }
+    }
+ 
+    // Use case 4: Retrieve values from the Workflow token.
+    
+    // To get the name of the last node which set the "ERROR" flag in the WorkflowToken:
+    
+    List<NodeValueEntry> errorNodeValueList = token.getAll("ERROR");
```

Comment:
I think this change may not have gotten pushed. Still shows as ``NodeValueEntry``.